src/boot/efi/pe.c

```diff
@@ -209,8 +209,14 @@ static void pe_locate_sections(
                         /* At this time, the sizes and offsets have been validated. Store them away */
                         sections[i] = (PeSectionVector) {
                                 .memory_size = j->VirtualSize,
-                                .file_offset = j->PointerToRawData,
                                 .memory_offset = j->VirtualAddress,
+                                /* VirtualSize can be bigger than SizeOfRawData when the section requires
+                                 * uninitialized data. It can also be smaller than SizeOfRawData when there's
+                                 * no need for uninitialized data as SizeOfRawData is aligned to
+                                 * FileAlignment and VirtualSize isn't. The actual data that's read from disk
+                                 * is the minimum of these two fields. */
+                                .file_size = MIN(j->SizeOfRawData, j->VirtualSize),
```

Comment:
VirtualSize we validated with two overflow checks above, SizeOfRawData should really receive similar treatment, before we store it away and make use of it

Reply:
(in particular because of the 64bit vs. 32it thing)

Reply:
Changed the `VirtualSize` overflow check against `PointerToRawData` to check `SizeOfRawData` instead since we don't use `PointerToRawData` together with `VirtualSize` anymore now.